hazelcast/src/test/java/com/hazelcast/topic/TopicTest.java

```diff
@@ -136,6 +140,108 @@ public void run() {
         });
     }
 
+    @Test
+    public void testTopicPublishAsync() throws Exception {
+        final String randomName = "testTopicPublishAsync" + generateRandomString(5);
+        final AtomicInteger count = new AtomicInteger(0);
+
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(1);
+        HazelcastInstance instance = factory.newHazelcastInstance();
+        ITopic<String> topic = instance.getTopic(randomName);
+        topic.addMessageListener(new MessageListener<String>() {
+
+            @Override
+            public void onMessage(Message<String> message) {
+                count.incrementAndGet();
+            }
+        });
+
+        final CompletableFuture<Void> f = topic.publishAsync("TestMessage").toCompletableFuture();
```

Comment:
You can use `assertCompletesEventually` to avoid blocking the test forever in case it never completes, same in other test methods doing the same.

Reply:
Done